### PR TITLE
[dev] Move BPF VM creation into image build

### LINKED ISSUE #4945: Move qemu image build to Docker image rather than the init tasks
Right now we're building the agent-smith qemu image for every prebuild. Instead we should be building it once during the image build. This way, prebuilds will be considerably quicker.

### PR DESCRIPTION
**Pull request by geropl**
Fixes #4945.

Also, makes the `img` even smaller by changing the format from `raw` to `qcow2`.
image

### Test
 - start a workspace on this branch
 - see that the VM is still started up on workspace start, but without the init phase/log

### Future work:
 - build time is still quite high if we need to rebuild the `dev` image. Converging all in-VM steps into a single script instead of starting/stopping it for every step should help.

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4959#" title="Author self-approved">geropl</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **rl-gitpod** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @rl-gitpod` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [geropl]
- **[components/ee/agent-smith/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ee/agent-smith/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4959#" title="Author self-approved">geropl</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **csweichel** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @csweichel` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [geropl]
- **[components/ee/agent-smith/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ee/agent-smith/OWNERS)**
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by geropl** [comment]
@aledbf Would be nice if you found time for a review today or tomorrow :pray:

File: components/ee/agent-smith/scripts/qemu.sh [hunk 1]
```diff
@@ -12,15 +12,14 @@ fi
 
 set -euo pipefail
 
-script_dirname="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
-outdir="${script_dirname}/_output"
+outdir="/root/_output"
 
 
 rm -Rf ~/.ssh
 cp -r "${outdir}/.ssh" ~/.ssh
```
Comment (aledbf) [inline components/ee/agent-smith/scripts/qemu.sh:19]: I get this error when I start a workspace:
```shell
gitpod /workspace/gitpod $ leeway run components/ee/agent-smith:qemu
cp: cannot stat '/root/_output/.ssh': Permission denied
```

Comment (aledbf) [inline components/ee/agent-smith/scripts/qemu.sh:19]: `sudo` solves the issue

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4959#" title="Author self-approved">geropl</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **csweichel** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @csweichel` in a comment when ready.

Associated issue: *#4945*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [geropl]
- **[components/ee/agent-smith/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ee/agent-smith/OWNERS)**
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 1819057cb06f306f2fb98a436965d5337bf1b744</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4959#issuecomment-887647411" title="LGTM">aledbf</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/4959#" title="Author self-approved">geropl</a>*

Associated issue: *#4945*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [aledbf,geropl]
- ~~[components/ee/agent-smith/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ee/agent-smith/OWNERS)~~ [aledbf]
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by csweichel** [comment]
The VM still works as indented 🚀 

image

I also see the `.ssh` permission issue though

**Comment by aledbf** [comment]
> I also see the .ssh permission issue though

Right, I fixed the copy but not the permissions

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

**Comment by aledbf** [comment]
@csweichel please try again

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: d04f80ec4ef1c0ed7fa0e6e637ef8feac2e49034</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/4959#issuecomment-887647411" title="LGTM">aledbf</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/4959#issuecomment-888042464" title="LGTM">csweichel</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/4959#" title="Author self-approved">geropl</a>*

Associated issue: *#4945*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [aledbf,csweichel,geropl]
- ~~[components/ee/agent-smith/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ee/agent-smith/OWNERS)~~ [aledbf,csweichel]
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [csweichel,geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**